### PR TITLE
Azure Elastic Agent VM extension

### PR DESCRIPTION
**Pull request by narph**
## What does this PR do?

Azure Elastic Agent VM extension

## Why is it important?

The Elastic Agent extension will provide the ability to collect metrics and logs from the target VM it is installed on and send this data to the user’s existing Elastic Cloud account. 

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

Docker files

## Related issues

- 

Will enrich description during progress

### CODE REVIEW

**Comment by narph** [comment]
@masci , @andresrc , @ph , open to suggestion where the implementation for the VM extension should reside (which repo). I have temporarily added in the same location as the elastic agent since they are related.

**Comment by andresrc** [comment]
I would say let's leave it here for now.

**Comment by exekias** [comment]
@narph I understand this is ready for review, right? it would be good to have a README.md file in the folder providing some context and pointers.

Will have a look to the PR 👁️

File: x-pack/azure-vm-extension/handler/linux/config_update.sh [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+   fi
+   log "INFO" "[Uninstall_Old_ElasticAgent_DEB_RPM] removing Elastic Agent directories"
+   sudo systemctl stop elastic-agent
+   sudo systemctl disable elastic-agent
+   sudo rm -rf /usr/share/elastic-agent
+   sudo rm -rf /etc/elastic-agent
+   sudo rm -rf /var/lib/elastic-agent
+   sudo rm -rf /usr/bin/elastic-agent
+   sudo systemctl daemon-reload
+   sudo systemctl reset-failed
```
Comment (exekias) [inline x-pack/azure-vm-extension/handler/linux/config_update.sh:68]: any reason why not relying on dpkg here?

File: x-pack/azure-vm-extension/handler/linux/enable.sh [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+      log "INFO" "[Install_ElasticAgent_RPM] download location - $DOWNLOAD_URL"
+      wget --retry-connrefused --waitretry=1 "$DOWNLOAD_URL" -O $PACKAGE
+      EXIT_CODE=$?
+      if [[ $EXIT_CODE -ne 0 ]]; then
+        log "ERROR" "[Install_ElasticAgent_RPM] error downloading Elastic Agent $STACK_VERSION"
+        return $EXIT_CODE
+      fi
+      log "INFO" "[Install_ElasticAgent_RPM] downloaded Elastic Agent $STACK_VERSION"
+      write_status "$name" "$first_operation" "transitioning" "$message" "$sub_name" "success" "Elastic Agent package has been downloaded"
+      #checkShasum $PACKAGE $SHASUM
```
Comment (exekias) [inline x-pack/azure-vm-extension/handler/linux/enable.sh:98]: Do you plan to enable this check?

**Comment by narph** [comment]
@exekias, this will live somewhere else, will close

**Pull request closed without merging**